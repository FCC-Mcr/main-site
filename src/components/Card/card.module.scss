@import "../../styles/config/colour.scss";

.card {
  background: $white;
  color: $black;
  border-radius: 5px;
}
