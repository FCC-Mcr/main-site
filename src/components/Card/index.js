import React from "react"
import styles from "./card.module.scss"

/** Card wrapper component
 * @param { string } className class to add to className string
 * @param { Number } height determines box-shadow strength
 */
const Card = ({ children, className, shadowSize = "large" }) => {
  const shadowClass = shadowSize ? `shadow-${shadowSize}` : ""

  return (
    <div className={`${styles.card} ${className} ${shadowClass}`}>
      {children}
    </div>
  )
}
export default Card
