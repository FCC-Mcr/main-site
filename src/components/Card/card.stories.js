import React from "react"

import { storiesOf } from "@storybook/react"

import Card from "./"

storiesOf("Card").add(
  "wibble",
  () => (
    <Card>
      <div>will render content in a basic card component.</div>
    </Card>
  ),
  {
    notes: "hi im notes on the cards",
  }
)
