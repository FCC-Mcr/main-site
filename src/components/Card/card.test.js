import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import Card from "./"

describe("Basic Card Component functionality", () => {
  // TODO: havent applied border radius tests as im not sure the exact use case (should be done with class?)

  test("Renders empty with defaults when called with no props", () => {
    const { container } = render(<Card />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test("Renders children", () => {
    const { container, getAllByText } = render(
      <Card>
        <div>child1</div>
        <div>child2</div>
      </Card>
    )
    expect(getAllByText("child1")).toHaveLength(1)
    expect(getAllByText("child2")).toHaveLength(1)
    expect(getAllByText("child", { exact: false })).toHaveLength(2)
    expect(container.firstChild).toMatchSnapshot()
  })

  test("Appends className prop to class", () => {
    const { container } = render(<Card className="addedText" />)
    expect(container.firstChild.classList.contains("addedText")).toBe(true)
    expect(container.firstChild).toMatchSnapshot()
  })

  test("Appends className for shadow based on height prop", () => {
    const { container } = render(<Card height={2} />)
    expect(container.firstChild.classList.contains("u-shadow--2")).toBe(true)
    expect(container.firstChild).toMatchSnapshot()
  })
})
