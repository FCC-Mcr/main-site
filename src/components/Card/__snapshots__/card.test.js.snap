// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Card Component functionality Appends className for shadow based on height prop 1`] = `
<div
  class="card  u-shadow--2"
  style="border-radius: 10px;"
/>
`;

exports[`Basic Card Component functionality Appends className prop to class 1`] = `
<div
  class="card addedText "
  style="border-radius: 10px;"
/>
`;

exports[`Basic Card Component functionality Renders children 1`] = `
<div
  class="card  "
  style="border-radius: 10px;"
>
  <div>
    child1
  </div>
  <div>
    child2
  </div>
</div>
`;

exports[`Basic Card Component functionality Renders empty with defaults when called with no props 1`] = `
<div
  class="card  "
  style="border-radius: 10px;"
/>
`;
